src/istore.h

```diff
@@ -150,12 +150,21 @@ int is_int32_arr_comp(const void *a, const void *b);
     } while (0)
 
 #define SAMESIGN(a, b) (((a) < 0) == ((b) < 0))
+
+#if defined(HAVE__BUILTIN_OP_OVERFLOW)
 #define INTPL(_a, _b, _r)                                                                                              \
-    do                                                                                                                 \
-    {                                                                                                                  \
-        _r = _a + _b;                                                                                                  \
-        if (SAMESIGN(_a, _b) && !SAMESIGN(_r, _a))                                                                     \
+    do {                                                                                                               \
+        if (__builtin_add_overflow(_a, _b, &_r))                                                                       \
             ereport(ERROR, (errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE), errmsg("integer out of range")));             \
     } while (0)
+#else
+#define INTPL(_a, _b, _r)                                                                                              \
```

Comment:
I think you can reuse `pg_add_s32_overflow()` and `pg_add_s64_overflow()` and delete `INTPL()` macros for good.

```
if (pg_add_s32_overflow(pairs[index].val, clamp_sum, &clamp_sum))
    ereport(ERROR, (errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE), errmsg("integer out of range")));
```

Reply:
That was my intention, but then CI reported failures. Turned out those functions were introduced in pg11. We gotta keep supporting older version (at least up to pg10).